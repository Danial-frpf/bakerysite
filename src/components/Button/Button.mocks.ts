import { IButton } from ".";

const small: IButton = {
    label: "Small",
    size: "sm",
};

const medium: IButton = {
    label: "Medium",
    size: "md",
};

const large: IButton = {
    label: "Large",
    size: "lg",
};

const extraLarge: IButton = {
    label: "extraLarge",
    size: "xl",
};

export const mockButtonProps = {
    small,
    medium,
    large,
    extraLarge,
};
