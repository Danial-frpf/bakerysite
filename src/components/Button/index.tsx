import Label from "../Label";
import styles from "./styles.module.css";

export interface IButton {
    label: string;
    size?: "sm" | "md" | "lg" | "xl";
    type?: "submit" | "button";
    props?: React.DetailedHTMLProps<
        React.ButtonHTMLAttributes<HTMLButtonElement>,
        HTMLButtonElement
    >;
}

const Button: React.FC<IButton> = ({
    label,
    size = "md",
    type = "button",
    props,
}) => {
    let sizeStyle = "";

    switch (size) {
        case "sm":
            sizeStyle = styles.sm;
            break;
        case "md":
            sizeStyle = styles.md;
            break;
        case "lg":
            sizeStyle = styles.lg;
            break;
        case "xl":
            sizeStyle = styles.xl;
            break;
        default:
            break;
    }

    return (
        <button
            type={type}
            data-testid="Button"
            {...props}
            className={`${styles.container} ${sizeStyle} ${props?.className}`}
        >
            <Label label={label} size={size} />
        </button>
    );
};

export default Button;
