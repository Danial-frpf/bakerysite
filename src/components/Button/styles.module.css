.container {
    background-color: var(--secondary-orange-40);
    border: 2px solid var(--main-800);
    transition: scale 500ms ease-in-out;
}

.container:hover {
    background-color: var(--secondary-yellow-40);
    border-color: var(--secondary-orange);
}

.container:active {
    background-color: var(--secondary-blue-80);
    border-color: var(--secondary-orange);
    scale: 0.8;
}

.container span:hover {
    color: var(--secondary-orange);
}

.sm {
    border-radius: 4px;
    padding: 0.5vh 0.5vw;
    margin-bottom: 0.5vh;
}

.md {
    border-radius: 8px;
    padding: 0.5vh 1vw;
    margin-bottom: 1vh;
}

.lg {
    border-radius: 8px;
    padding: 1vh 1vw;
    margin-bottom: 1vh;
}

.xl {
    border-radius: 16px;
    margin-bottom: 2vh;
    padding: 1vh 2vw;
}
