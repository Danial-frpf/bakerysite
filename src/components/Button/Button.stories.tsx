import { ComponentStory, ComponentMeta } from "@storybook/react";
import Button, { IButton } from ".";
import { mockButtonProps } from "./Button.mocks";

export default {
    title: "templates/Button",
    component: Button,
    // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
    argTypes: {},
} as ComponentMeta<typeof Button>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Small = Template.bind({});
export const Medium = Template.bind({});
export const Large = Template.bind({});
export const ExtraLarge = Template.bind({});

// More on args: https://storybook.js.org/docs/react/writing-stories/args

Small.args = {
    ...mockButtonProps.small,
} as IButton;

Medium.args = {
    ...mockButtonProps.medium,
} as IButton;

Large.args = {
    ...mockButtonProps.large,
} as IButton;

ExtraLarge.args = {
    ...mockButtonProps.extraLarge,
} as IButton;
