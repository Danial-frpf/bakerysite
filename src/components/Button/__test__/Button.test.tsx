import { createRoot } from "react-dom/client";
import { act } from "react-dom/test-utils";
import Button from "..";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(<Button label="Button"></Button>);
    });
});

it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(<Button label="Button" size="lg"></Button>);
    });
});

it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(<Button label="Button" size="sm"></Button>);
    });
});

it("renders Button correctly", () => {
    const { getByTestId } = render(<Button label="Button"></Button>);
    expect(getByTestId("Button")).toHaveTextContent("Button");
});

it("renders Button correctly", () => {
    const { getByTestId } = render(<Button label="Button" size="lg"></Button>);
    expect(getByTestId("Button")).toHaveTextContent("Button");
});

it("renders Button correctly", () => {
    const { getByTestId } = render(<Button label="Button" size="sm"></Button>);
    expect(getByTestId("Button")).toHaveTextContent("Button");
});

it("matches snapshot", () => {
    const tree = renderer.create(<Button label="Button"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<Button label="Button" size="lg"></Button>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<Button label="Button" size="sm"></Button>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
