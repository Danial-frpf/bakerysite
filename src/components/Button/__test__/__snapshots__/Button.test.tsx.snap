// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="container md undefined"
  data-testid="Button"
  type="button"
>
  <h4
    data-testid="Label"
  >
    <span>
      Button
    </span>
  </h4>
</button>
`;

exports[`matches snapshot 2`] = `
<button
  className="container lg undefined"
  data-testid="Button"
  type="button"
>
  <h2
    data-testid="Label"
  >
    <span>
      Button
    </span>
  </h2>
</button>
`;

exports[`matches snapshot 3`] = `
<button
  className="container sm undefined"
  data-testid="Button"
  type="button"
>
  <p
    data-testid="Label"
  >
    <span>
      Button
    </span>
  </p>
</button>
`;

exports[`matches snapshot 4`] = `
<button
  className="container xl undefined"
  data-testid="Button"
  type="button"
>
  <h1
    data-testid="Label"
  >
    <span>
      Button
    </span>
  </h1>
</button>
`;
